packages/Interface/src/DTK_C_API.cpp

```diff
@@ -0,0 +1,433 @@
+        dtk->_registry->setBoundingVolumeListDataFunction(
+            BoundingVolumeListDataFunctionWrapper, data );
+        break;
+    case DTK_POLYHEDRON_LIST_SIZE_FUNCTION:
+        dtk->_registry->setPolyhedronListSizeFunction(
+            PolyhedronListSizeFunctionWrapper, data );
+        break;
+    case DTK_POLYHEDRON_LIST_DATA_FUNCTION:
+        dtk->_registry->setPolyhedronListDataFunction(
+            PolyhedronListDataFunctionWrapper, data );
+        break;
+    case DTK_CELL_LIST_SIZE_FUNCTION:
+        dtk->_registry->setCellListSizeFunction( CellListSizeFunctioWrapper,
+                                                 data );
+        break;
+    case DTK_CELL_LIST_DATA_FUNCTION:
+        dtk->_registry->setCellListDataFunction( CellListDataFunctionWrapper,
+                                                 data );
+        break;
+    case DTK_MIXED_TOPOLOGY_CELL_LIST_SIZE_FUNCTION:
+        dtk->_registry->setMixedTopologyCellListSizeFunction(
+            MixedTopologyCellListSizeFunctionWrapper, data );
+        break;
+    case DTK_MIXED_TOPOLOGY_CELL_LIST_DATA_FUNCTION:
+        dtk->_registry->setMixedTopologyCellListDataFunction(
+            MixedTopologyCellListDataFunctionWrapper, data );
+        break;
+    case DTK_BOUNDARY_SIZE_FUNCTION:
+        dtk->_registry->setBoundarySizeFunction( BoundarySizeFunctionWrapper,
+                                                 data );
+        break;
+    case DTK_BOUNDARY_DATA_FUNCTION:
+        dtk->_registry->setBoundaryDataFunction( BoundaryDataFunctionWrapper,
+                                                 data );
+        break;
+    case DTK_DOF_MAP_SIZE_FUNCTION:
+        dtk->_registry->setDOFMapSizeFunction( DOFMapSizeFunctionWrapper,
+                                               data );
+        break;
+    case DTK_DOF_MAP_DATA_FUNCTION:
+        dtk->_registry->setDOFMapDataFunction( DOFMapDataFunctionWrapper,
+                                               data );
+        break;
+    case DTK_MIXED_TOPOLOGY_DOF_MAP_SIZE_FUNCTION:
+        dtk->_registry->setMixedTopologyDOFMapSizeFunction(
+            MixedTopologyDOFMapSizeFunctionWrapper, data );
+        break;
+    case DTK_MIXED_TOPOLOGY_DOF_MAP_DATA_FUNCTION:
+        dtk->_registry->setMixedTopologyDOFMapDataFunction(
+            MixedTopologyDOFMapDataFunctionWrapper, data );
+        break;
+    case DTK_FIELD_SIZE_FUNCTION:
+        dtk->_registry->setFieldSizeFunction( FieldSizeFunctionWrapper, data );
+        break;
+    case DTK_PULL_FIELD_DATA_FUNCTION:
+        dtk->_registry->setPullFieldDataFunction(
+            PullFieldDataFunctionWrapper<double>, data );
+        break;
+    case DTK_PUSH_FIELD_DATA_FUNCTION:
+        dtk->_registry->setPushFieldDataFunction(
+            PushFieldDataFunctionWrapper<double>, data );
+        break;
+    case DTK_EVALUATE_FIELD_FUNCTION:
+        dtk->_registry->setEvaluateFieldFunction(
+            EvaluateFieldFunctionWrapper<double>, data );
```

Comment:
Don't you get a warning for not having a `default` case? I think it would be good to add one that throws. This function is going to be visible by the user right?